File: 128-longest-consecutive-sequence/128-longest-consecutive-sequence.java
class Solution {
    public int longestConsecutive(int[] nums) {
        HashSet<Integer> numSet = new HashSet<>();
        for(int num : nums)
            numSet.add(num);
        
        int ans = 0;
        for(int num : nums){
            int count = 0;
            int nnum = num;
            while(numSet.contains(nnum)){
                numSet.remove(nnum);
                nnum--;
                count++;
            }
            
            nnum = num + 1;
            while(numSet.contains(nnum)){
                numSet.remove(nnum);
                nnum++;
                count++;
            }
            ans = Math.max(ans, count);
        }
        return ans;
    }
}

File: 128-longest-consecutive-sequence/128-longest-consecutive-sequence.py
class Solution:
    def longestConsecutive(self, nums: List[int]) -> int:
        num_set = set(nums)
        
        ans = 0
        for num in nums:
            count = 0
            nnum = num
            while nnum in num_set:
                num_set.remove(nnum)
                count += 1
                nnum -= 1
            
            nnum = num + 1
            while nnum in num_set:
                num_set.remove(nnum)
                count += 1
                nnum += 1
            ans = max(ans, count)
        return ans